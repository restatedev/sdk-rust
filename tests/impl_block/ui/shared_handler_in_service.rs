use restate_sdk::prelude::*;

struct SharedHandlerInService;

#[restate_sdk::service]
impl SharedHandlerInService {
    #[handler(shared)]
    async fn my_handler(&self, _ctx: Context<'_>) -> HandlerResult<()> {
        Ok(())
    }
}

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt::init();
    HttpServer::new(
        Endpoint::builder()
            .with_service(SharedHandlerInService.serve())
            .build(),
    )
        .listen_and_serve("0.0.0.0:9080".parse().unwrap())
        .await;
}
