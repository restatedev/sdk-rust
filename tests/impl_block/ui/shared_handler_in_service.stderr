error: Service handlers cannot be annotated with #[handler(shared)]
 --> tests/ui/shared_handler_in_service.rs:7:15
  |
7 |     #[handler(shared)]
  |               ^^^^^^

error[E0599]: no method named `with_service` found for struct `restate_sdk::endpoint::Builder` in the current scope
  --> tests/ui/shared_handler_in_service.rs:18:14
   |
17 | /         Endpoint::builder()
18 | |             .with_service(SharedHandlerInService.serve())
   | |             -^^^^^^^^^^^^ method not found in `Builder`
   | |_____________|
   |

error[E0599]: no method named `serve` found for struct `SharedHandlerInService` in the current scope
  --> tests/ui/shared_handler_in_service.rs:18:50
   |
3  | struct SharedHandlerInService;
   | ----------------------------- method `serve` not found for this struct
...
18 |             .with_service(SharedHandlerInService.serve())
   |                                                  ^^^^^ method not found in `SharedHandlerInService`
